### DISCUSSION TITLE
how to tell @wordpress/scripts to register scripts fo in_footer position?

### DISCUSSION THREAD

**Discussion by stratboy** [WordPress/gutenberg#48647] [Q&A]
Hi, Iìm currently testing @wordpress/scripts to build blocks js, but I can see that its default is to register the scripts with default parameters, so that they will load into the head section. I'd like them to be in footer. Also I'd like to add defer property. Ho to hook into that? I used to use plain wp_register/enqueue_script, but this time it seems it's all done by @wordpress/scripts. So how to?

**Comment by gziolo** [comment]
There is a related issue opened https://github.com/WordPress/gutenberg/issues/46954. There is no simple way to do it today. You can still register a script yourself with `wp_register_script` and pass the script handle in the `block.json` file. The challenge is that you would need to tweak the default handling in `@wordpress/scripts` so it knows how to build the script.

**Reply by stratboy** [reply]
Thank you, but I ended up opting for good ol' and plain webpack configuration. No `@wordpress/scripts`. 
I also needed to do at least the following:

- defer the scripts. This is currently only possible with filters. It should be possible even via enqueue/register_script. 
- only load a block script when the block appears in the current post. This is also currently possible only with filters.

The 2 above force me to load scripts manually, and thus, @wordpress/scripts is not currently an option for me.
It would be great if wordpress devs concentrate also on the 2 above 'issues'. They should be available in more convenent ways than filters, in my opinion, nowadays.

Instead, I realized that placing scripts on the footer it's not that important after all, as long as it's possible to defer scripts.

**Reply by gziolo** [reply]
There is an active proposal [Enhancing the Scripts API with a loading strategy](https://make.wordpress.org/core/2022/12/09/enhancing-the-scripts-api-with-a-loading-strategy/). One of the options would be to use `defer` with scripts. It's on the 2023 roadmap for the Core Performance team in the [JavaScript & CSS section](https://make.wordpress.org/performance/roadmap-2023/#category-javascript-css).
